tests/components/minecraft_server/test_config_flow.py

```diff
@@ -87,54 +103,96 @@
-        DOMAIN, context={"source": SOURCE_USER}, data=USER_INPUT_PORT_TOO_SMALL
-    )
+    with patch(
+        "aiodns.DNSResolver.query", side_effect=aiodns.error.DNSError,
+    ):
+        result = await hass.config_entries.flow.async_init(
+            DOMAIN, context={"source": SOURCE_USER}, data=USER_INPUT_PORT_TOO_SMALL
+        )
 
-    assert result["type"] == RESULT_TYPE_FORM
-    assert result["errors"] == {"base": "invalid_port"}
+        assert result["type"] == RESULT_TYPE_FORM
+        assert result["errors"] == {"base": "invalid_port"}
 
 
 async def test_port_too_large(hass: HomeAssistantType) -> None:
     """Test error in case of a too large port."""
-    result = await hass.config_entries.flow.async_init(
-        DOMAIN, context={"source": SOURCE_USER}, data=USER_INPUT_PORT_TOO_LARGE
-    )
+    with patch(
+        "aiodns.DNSResolver.query", side_effect=aiodns.error.DNSError,
+    ):
+        result = await hass.config_entries.flow.async_init(
+            DOMAIN, context={"source": SOURCE_USER}, data=USER_INPUT_PORT_TOO_LARGE
+        )
 
-    assert result["type"] == RESULT_TYPE_FORM
-    assert result["errors"] == {"base": "invalid_port"}
+        assert result["type"] == RESULT_TYPE_FORM
+        assert result["errors"] == {"base": "invalid_port"}
 
 
 async def test_connection_failed(hass: HomeAssistantType) -> None:
     """Test error in case of a failed connection."""
-    with patch("mcstatus.server.MinecraftServer.ping", side_effect=OSError):
-        result = await hass.config_entries.flow.async_init(
-            DOMAIN, context={"source": SOURCE_USER}, data=USER_INPUT
-        )
+    with patch(
+        "aiodns.DNSResolver.query", side_effect=aiodns.error.DNSError,
+    ):
+        with patch("mcstatus.server.MinecraftServer.status", side_effect=OSError):
+            result = await hass.config_entries.flow.async_init(
+                DOMAIN, context={"source": SOURCE_USER}, data=USER_INPUT
+            )
 
-        assert result["type"] == RESULT_TYPE_FORM
-        assert result["errors"] == {"base": "cannot_connect"}
+            assert result["type"] == RESULT_TYPE_FORM
+            assert result["errors"] == {"base": "cannot_connect"}
+
+
+async def test_connection_succeeded_with_srv_record(hass: HomeAssistantType) -> None:
+    """Test config entry in case of a successful connection with a SRV record."""
+    with patch(
+        "aiodns.DNSResolver.query", return_value=SRV_RECORDS,
+    ):
+        with patch(
+            "mcstatus.server.MinecraftServer.status",
+            return_value=PingResponse(STATUS_RESPONSE_RAW),
+        ):
+            result = await hass.config_entries.flow.async_init(
+                DOMAIN, context={"source": SOURCE_USER}, data=USER_INPUT_SRV
+            )
+
+            assert result["type"] == RESULT_TYPE_CREATE_ENTRY
+            assert result["title"] == USER_INPUT_SRV[CONF_HOST]
+            assert result["data"][CONF_NAME] == USER_INPUT_SRV[CONF_NAME]
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.](https://app.codacy.com/manual/foreign-sub/home-assistant/pullRequest?prid=5232399)